src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -952,5 +953,373 @@ private CorInfoHelpFunc getNewArrHelper(CORINFO_CLASS_STRUCT_* arrayCls)
+                    // during the jitting of the call.
+                    // (Note: The generic lookup in R2R is performed by a call to a helper at runtime, not by
+                    // codegen emitted at crossgen time)
+
+                    Debug.Assert(!forceUseRuntimeLookup);
+                    pResult->codePointerOrStubLookup.constLookup = CreateConstLookupToSymbol(
+                        _compilation.NodeFactory.MethodEntrypoint(targetMethod)
+                        );
+                }
+                else
+                {
+                    ISymbolNode instParam = null;
+
+                    if (targetMethod.RequiresInstMethodDescArg())
+                    {
+                        instParam = _compilation.NodeFactory.MethodGenericDictionary(concreteMethod);
+                    }
+                    else if (targetMethod.RequiresInstMethodTableArg() || referencingArrayAddressMethod)
+                    {
+                        // Ask for a constructed type symbol because we need the vtable to get to the dictionary
+                        instParam = _compilation.NodeFactory.ConstructedTypeSymbol(concreteMethod.OwningType);
+                    }
+
+                    if (instParam != null)
+                    {
+                        pResult->instParamLookup = CreateConstLookupToSymbol(instParam);
+                    }
+
+                    pResult->codePointerOrStubLookup.constLookup = CreateConstLookupToSymbol(
+                        _compilation.NodeFactory.MethodEntrypoint(targetMethod)
+                        );
+                }
+
+                pResult->nullInstanceCheck = resolvedCallVirt;
+            }
+            else if (method.HasInstantiation)
+            {
+                // GVM Call Support
+                pResult->kind = CORINFO_CALL_KIND.CORINFO_VIRTUALCALL_LDVIRTFTN;
+                pResult->codePointerOrStubLookup.constLookup.accessType = InfoAccessType.IAT_VALUE;
+                pResult->nullInstanceCheck = true;
+
+                if (pResult->exactContextNeedsRuntimeLookup)
+                {
+                    ComputeLookup(ref pResolvedToken,
+                        GetRuntimeDeterminedObjectForToken(ref pResolvedToken),
+                        ReadyToRunHelperId.MethodHandle,
+                        ref pResult->codePointerOrStubLookup);
+                    Debug.Assert(pResult->codePointerOrStubLookup.lookupKind.needsRuntimeLookup);
+                }
+
+                // RyuJIT will assert if we report CORINFO_CALLCONV_PARAMTYPE for a result of a ldvirtftn
+                // We don't need an instantiation parameter, so let's just not report it. Might be nice to
+                // move that assert to some place later though.
+                targetIsFatFunctionPointer = true;
+            }
+            else
+            // In ReadyToRun, we always use the dispatch stub to call virtual methods
```

Comment:
This comment should have been in the `#if READYTORUN` block. It's R2R specific.

Reply:
Removed